### PR TITLE
Integrating AutoAtom functionality

### PR DESCRIPTION
**Pull request by src-code**
Bringing this over from grunt-atomizer... Note this is not BC, and will result in a major version bump

### CODE REVIEW

**Comment by yahoocla** [comment]
CLA is valid!

**Comment by src-code** [comment]
Need to fix the tests, and then add some docs....

**Comment by src-code** [comment]
Still needs some doc updates, please don't merge until that's added

File: bin/atomizer [hunk 1]
```diff
@@ -14,16 +14,30 @@ var path = require('path');
...
 var params = require('minimist')(process.argv.slice(2));
 
 if (process.argv.slice(2).length === 0 || params.help) {
-    var usage = ['usage:  ', process.title, '[-o|--outfile=<file>] [--help] configfile ...'].join(' ');
+    var usage = ['usage:  ', process.title, ' -c|--config=<file> [-o|--outfile=<file>] [--help] [--verbose] [<files-to-parse> ...]'].join(' ');
     console.log(usage);
     return;
 }
 
+function handleMergeArrays (a, b) {
```
Comment (renatoi) [inline bin/atomizer]: We should move this function to `utils` so we don't have duplicates.

File: src/atomizer.js [hunk 2]
```diff
@@ -11,54 +11,279 @@ var Absurd = require('absurd');
...
+var verbose = false;
 
-    var content;
+/**
+ * helper function to handle merging array of objects
+ * @param  {mixed} a Data of the first merge param
+ * @param  {mixed} b Data of the second merge param
+ * @return {mixed}   The merged object
+ */
+function handleMergeArrays (a, b) {
```
Comment (renatoi) [inline src/atomizer.js]: Just point out that we need to move this to `utils`.

**Comment by renatoi** [comment]
Other that, I think it looks good to me. We should update the docs to reflect how to use, specially programatically.

One feedback though. As a developer, I'd like to have an option to parse a file and get the css, either through chaining it or one call.

```javascript
var atomizer = require('atomizer');

// either this
atomizer.parse(src).getConfig().createCSS();

// or this
atomizer.atomify(src);
```

**Comment by src-code** [comment]
I don't know if chaining is realistic (since that'd require the APIs to change their return value, and maintain state), but I do like the idea of offering something like `atomize()` that wraps all these calls...

**Pull request merged**